### PR TITLE
Support personal message signing

### PR DESCRIPTION
**Pull request by yuliaveres**

### CODE REVIEW

File: Toshi/SOFAWebview/SOFAWebController.swift [hunk 1]
```diff
@@ -257,6 +291,16 @@ extension SOFAWebController: WKScriptMessageHandler {
             jsCallback(callbackId: callbackId, payload: payload)
         }
     }
+
+    private func presentPersonalMessageSignAlert(_ message: String, signHandler: @escaping (() -> Void)) {
+        let alert = UIAlertController(title: "\"CryptoKitties\" is requesting\nyour Signature", message: message, preferredStyle: .alert)
```
Comment (designatednerd) [inline Toshi/SOFAWebview/SOFAWebController.swift]: 🚨 non-localized user-facing strings 🚨

File: Toshi/SOFAWebview/SOFAWebController.swift [hunk 2]
```diff
@@ -186,6 +188,38 @@ extension SOFAWebController: WKScriptMessageHandler {
...
+            if let messageData = messageEncodedString.hexadecimalData, let decodedString = String(data: messageData, encoding: .utf8) {
+
+                DispatchQueue.main.async {
+                    self.presentPersonalMessageSignAlert(decodedString, signHandler: { [weak self] in
+                        let composedString = "\u{0019}Ethereum Signed Message:\n" + String(messageData.count) + decodedString
+
+                        if let resultData = composedString.data(using: .utf8) {
+                            var signature = "0x\(Cereal.shared.signWithWallet(hex: resultData.hexEncodedString()))"
+
+                            let index = signature.index(signature.startIndex, offsetBy: 130)
```
Comment (designatednerd) [inline Toshi/SOFAWebview/SOFAWebController.swift]: where are these numbers coming from?

Comment (yuliaveres) [reply Toshi/SOFAWebview/SOFAWebController.swift]: signature length - 2 last symbols :/

**Comment by designatednerd** [comment]
Is there any way to test this? It seems like something we'd want to make damned sure is working.

**Comment by yuliaveres** [comment]
you can search on `Debug`for "Personal Sign Tester" and sign a message. It should succeed

**Comment by designatednerd** [comment]
I meant in an automated test :smile:

**Comment by yuliaveres** [comment]
hmm 🤔  not sure right now

**Comment by tristan** [comment]
This is a bit time sensitive right now, and i think there might bit too much extra framework (mocking into the webview js pipeline) and/or rewriting needed to do good testing in this PR. I'd save it for another PR

**Comment by designatednerd** [comment]
Ok - definitely needs to be added though - I want to make sure people don't lose 😹

**Pull request merged**